## *********************************************************
## 
## File autogenerated for the cvt_ros package 
## by the dynamic_reconfigure package.
## Please do not edit.
## 
## ********************************************************/

##**********************************************************
## Software License Agreement (BSD License)
##
##  Copyright (c) 2008, Willow Garage, Inc.
##  All rights reserved.
##
##  Redistribution and use in source and binary forms, with or without
##  modification, are permitted provided that the following conditions
##  are met:
##
##   * Redistributions of source code must retain the above copyright
##     notice, this list of conditions and the following disclaimer.
##   * Redistributions in binary form must reproduce the above
##     copyright notice, this list of conditions and the following
##     disclaimer in the documentation and/or other materials provided
##     with the distribution.
##   * Neither the name of the Willow Garage nor the names of its
##     contributors may be used to endorse or promote products derived
##     from this software without specific prior written permission.
##
##  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
##  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
##  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
##  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
##  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
##  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
##  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
##  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
##  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
##  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
##  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
##  POSSIBILITY OF SUCH DAMAGE.
##**********************************************************/

from dynamic_reconfigure.encoding import extract_params

inf = float('inf')

config_description = {'upper': 'DEFAULT', 'lower': 'groups', 'srcline': 233, 'name': 'Default', 'parent': 0, 'srcfile': '/opt/ros/groovy/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'cstate': 'true', 'parentname': 'Default', 'class': 'DEFAULT', 'field': 'default', 'state': True, 'parentclass': '', 'groups': [], 'parameters': [{'srcline': 259, 'description': 'Shutter Control Mode', 'max': 2, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/groovy/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'shutter_mode', 'edit_method': "{'enum_description': 'An enum to set feature modes', 'enum': [{'srcline': 9, 'description': 'Automatic mode', 'srcfile': '/home/kloses/Develop/catkin_ws/src/cvt_ros/cvt_ros/cfg/ChameleonSettings.cfg', 'cconsttype': 'const int', 'value': 0, 'ctype': 'int', 'type': 'int', 'name': 'AUTO'}, {'srcline': 10, 'description': 'Manual mode', 'srcfile': '/home/kloses/Develop/catkin_ws/src/cvt_ros/cvt_ros/cfg/ChameleonSettings.cfg', 'cconsttype': 'const int', 'value': 1, 'ctype': 'int', 'type': 'int', 'name': 'MANUAL'}, {'srcline': 11, 'description': 'One Push Automatic mode', 'srcfile': '/home/kloses/Develop/catkin_ws/src/cvt_ros/cvt_ros/cfg/ChameleonSettings.cfg', 'cconsttype': 'const int', 'value': 2, 'ctype': 'int', 'type': 'int', 'name': 'ONE_PUSH_AUTO'}]}", 'default': 1, 'level': 0, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'Exposure Control Mode', 'max': 2, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/groovy/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'exposure_mode', 'edit_method': "{'enum_description': 'An enum to set feature modes', 'enum': [{'srcline': 9, 'description': 'Automatic mode', 'srcfile': '/home/kloses/Develop/catkin_ws/src/cvt_ros/cvt_ros/cfg/ChameleonSettings.cfg', 'cconsttype': 'const int', 'value': 0, 'ctype': 'int', 'type': 'int', 'name': 'AUTO'}, {'srcline': 10, 'description': 'Manual mode', 'srcfile': '/home/kloses/Develop/catkin_ws/src/cvt_ros/cvt_ros/cfg/ChameleonSettings.cfg', 'cconsttype': 'const int', 'value': 1, 'ctype': 'int', 'type': 'int', 'name': 'MANUAL'}, {'srcline': 11, 'description': 'One Push Automatic mode', 'srcfile': '/home/kloses/Develop/catkin_ws/src/cvt_ros/cvt_ros/cfg/ChameleonSettings.cfg', 'cconsttype': 'const int', 'value': 2, 'ctype': 'int', 'type': 'int', 'name': 'ONE_PUSH_AUTO'}]}", 'default': 1, 'level': 0, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'Gain Control Mode', 'max': 2, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/groovy/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'gain_mode', 'edit_method': "{'enum_description': 'An enum to set feature modes', 'enum': [{'srcline': 9, 'description': 'Automatic mode', 'srcfile': '/home/kloses/Develop/catkin_ws/src/cvt_ros/cvt_ros/cfg/ChameleonSettings.cfg', 'cconsttype': 'const int', 'value': 0, 'ctype': 'int', 'type': 'int', 'name': 'AUTO'}, {'srcline': 10, 'description': 'Manual mode', 'srcfile': '/home/kloses/Develop/catkin_ws/src/cvt_ros/cvt_ros/cfg/ChameleonSettings.cfg', 'cconsttype': 'const int', 'value': 1, 'ctype': 'int', 'type': 'int', 'name': 'MANUAL'}, {'srcline': 11, 'description': 'One Push Automatic mode', 'srcfile': '/home/kloses/Develop/catkin_ws/src/cvt_ros/cvt_ros/cfg/ChameleonSettings.cfg', 'cconsttype': 'const int', 'value': 2, 'ctype': 'int', 'type': 'int', 'name': 'ONE_PUSH_AUTO'}]}", 'default': 1, 'level': 0, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'White-Bal. Mode', 'max': 2, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/groovy/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'white_balance_mode', 'edit_method': "{'enum_description': 'An enum to set feature modes', 'enum': [{'srcline': 9, 'description': 'Automatic mode', 'srcfile': '/home/kloses/Develop/catkin_ws/src/cvt_ros/cvt_ros/cfg/ChameleonSettings.cfg', 'cconsttype': 'const int', 'value': 0, 'ctype': 'int', 'type': 'int', 'name': 'AUTO'}, {'srcline': 10, 'description': 'Manual mode', 'srcfile': '/home/kloses/Develop/catkin_ws/src/cvt_ros/cvt_ros/cfg/ChameleonSettings.cfg', 'cconsttype': 'const int', 'value': 1, 'ctype': 'int', 'type': 'int', 'name': 'MANUAL'}, {'srcline': 11, 'description': 'One Push Automatic mode', 'srcfile': '/home/kloses/Develop/catkin_ws/src/cvt_ros/cvt_ros/cfg/ChameleonSettings.cfg', 'cconsttype': 'const int', 'value': 2, 'ctype': 'int', 'type': 'int', 'name': 'ONE_PUSH_AUTO'}]}", 'default': 0, 'level': 0, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'Shutter Time (ms)', 'max': 200.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/groovy/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'shutter_value', 'edit_method': '', 'default': 5.0, 'level': 0, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'Exposure Value (IL)', 'max': 2.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/groovy/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'exposure_value', 'edit_method': '', 'default': 0.8, 'level': 0, 'min': -2.0, 'type': 'double'}, {'srcline': 259, 'description': 'Gain Value (%)', 'max': 20.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/groovy/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'gain_value', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'White Balance Value (U-B)', 'max': 1024, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/groovy/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'white_balance_ub_value', 'edit_method': '', 'default': 160, 'level': 0, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'White Balance Value (V-R)', 'max': 1024, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/groovy/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'white_balance_vr_value', 'edit_method': '', 'default': 160, 'level': 0, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'Automatic Shutter Regulation', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/groovy/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'useAutoShutter', 'edit_method': '', 'default': False, 'level': 0, 'min': False, 'type': 'bool'}, {'srcline': 259, 'description': 'Trigger a frame manually', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/groovy/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'trigger', 'edit_method': '', 'default': False, 'level': 0, 'min': False, 'type': 'bool'}], 'type': '', 'id': 0}

min = {}
max = {}
defaults = {}
level = {}
type = {}
all_level = 0

#def extract_params(config):
#    params = []
#    params.extend(config['parameters'])    
#    for group in config['groups']:
#        params.extend(extract_params(group))
#    return params

for param in extract_params(config_description):
    min[param['name']] = param['min']
    max[param['name']] = param['max']
    defaults[param['name']] = param['default']
    level[param['name']] = param['level']
    type[param['name']] = param['type']
    all_level = all_level | param['level']

ChameleonSettings_AUTO = 0
ChameleonSettings_MANUAL = 1
ChameleonSettings_ONE_PUSH_AUTO = 2
